src/main/java/com/csfrequency/firebase/devicestore/FirebaseDeviceStore.java

```diff
@@ -145,18 +153,24 @@ public void unsubscribe() {
     private void deleteDevice(final String userId) {
         final DocumentReference docRef = userRef(userId);
 
+        // TODO: This code might never have a chance to run if a user calls "signOut" and then
+        // exists out of the client. If this cleanup is required, then signOut() will have to return
+        // a Task.
         firestore.runTransaction(new Transaction.Function<Void>() {
             @Override
             public Void apply(Transaction transaction) throws FirebaseFirestoreException {
                 DocumentSnapshot doc = transaction.get(docRef);
 
+                // TODO: Would it be possible to achieve the same with FieldValue.arrayRemove()?
```

Comment:
After reviewing the iOS version of this, I'm wondering if we could make `devices` into a map instead of an array.  See https://github.com/CSFrequency/firebase-device-store-ios-sdk/pull/2/files#diff-915f072c5b986ba8886fbdc02a627c9dR128

I think this would greatly simplify the code.

Reply:
As part of the PR, I have switched to maps, and updated this method and `signOut` to return a `Task`